src/protocols/user_directed_commissioning/UserDirectedCommissioningServer.cpp

```diff
@@ -56,12 +56,12 @@ void UserDirectedCommissioningServer::OnMessageReceived(const Transport::PeerAdd
 
     ChipLogProgress(AppServer, "IdentityDeclaration DataLength()=%" PRIu32, static_cast<uint32_t>(msg->DataLength()));
 
-    uint8_t udcPayload[IdentificationDeclaration::kUdcTLVDataMaxBytes];
+    uint8_t udcPayload[IdentificationDeclaration::kUdcTLVDataMaxBytes] = {};
```

Comment:
This is about null (character) terminators, not nullptr, right?

Also: while this change is of course correct in the sense of "don't read un-unitialized data from the buffer", if there are things violating a SHALL-level requirement in the spec, those things should either fail cert, or fail to interoperate with things that are enforcing the spec requirements, or ideally both....

Reply:
> Is this something that we can do as a follow-up?

Yes.